repoaudit/repoaudit/__init__.py

```diff
@@ -53,19 +58,58 @@ def apt(recursive: bool, url: str, dists: str) -> None:
     else:
         dist_set = None
 
-    for repo_url in urls:
-        check_apt_repo(repo_url, dist_set)
+    if pubkeys:
+        try:
+            gpg = initialize_gpg(pubkeys.split(","))
+        except HTTPError as e:
+            raise click.ClickException(
+                f"{e}\n"
+                "Please check the url for the public key"
+            )
+    else:
+        gpg = None
+
+    errors = RepoErrors()
+    try:
+        for repo_url in urls:
+            check_apt_repo(repo_url, dist_set, gpg, errors)
+    except KeyboardInterrupt:
+        pass
+
+    destroy_gpg(gpg)
+    output_result(errors, output)
 
 
 @main.command()
 @recursive_option
 @click.argument("url")
-def yum(recursive: bool, url: str) -> None:
+@file_option
+@pubkey_option
+def yum(recursive: bool, url: str, output: str, pubkeys: str) -> None:
     """Validate a yum repository at URL."""
     if recursive:
-        urls = _get_repo_urls(url)
+        urls = get_repo_urls(url)
     else:
         urls = [url]
 
-    for repo_url in urls:
-        check_yum_repo(repo_url)
+    if pubkeys:
+        try:
+            gpg = initialize_gpg(pubkeys.split(","))
+        except HTTPError as e:
+            raise click.ClickException(
+                f"{e}\n"
+                "Please check the url for the public key"
+            )
+    else:
+        gpg = None
```

Comment:
This if/else could probably be moved into a function and shared with the apt command to make the code more DRY.

Reply:
Moved it to a separate function